src/Form/Fields.php

```diff
@@ -50,18 +60,22 @@ public function defaults(): array
 	 */
 	public function errors(): array
 	{
-		$errors = [];
+		if ($this->errors !== null) {
+			return $this->errors;
+		}
+
+		$this->errors = [];
 
 		foreach ($this->data as $name => $field) {
 			if ($field->errors() !== []) {
-				$errors[$name] = [
+				$this->errors[$name] = [
 					'label'   => $field->label(),
 					'message' => $field->errors()
```

Comment:
`Suggestion`: Instead of running `$field->errors()` twice, write it first to a variable, then check the variable for `!== []` and use it in the array.

Reply:
I'm sorry, I don't think I can follow :)

Reply:
You are currently calling `$field->errors()` in the if statement and then in the array assignment again.

I think it would be better to be

```php
$errors = $field->errors():
if ($errors !== []) {
    $this->errors[$name] = [
        'label' => $field->label(),
        'errors' => $errors
    ]
}
```

Reply:
Oh, now I got you. Yes, that definitely makes sense.